bin/varnishd/cache/cache_backend_probe.c

```diff
@@ -658,12 +680,18 @@ VBP_Control(const struct backend *be, int enable)
 
 	Lck_Lock(&vbp_mtx);
 	if (enable) {
+		// XXX next two assertions are WRONG, see #4108 - WIP
+		assert(vt->state == vbp_state_cold);
 		assert(vt->heap_idx == VBH_NOIDX);
 		vt->due = VTIM_real();
 		vbp_heap_insert(vt);
+		vt->state = vbp_state_scheduled;
 	} else {
+		assert(vt->state == vbp_state_scheduled ||
+		       vt->state == vbp_state_running);
```

Comment:
Wrong indentation.

Reply:
depends on your definition of "right". But anyway...